paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/compact/UnawareBucketCompactionTopologyBuilder.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.paimon.utils.SerializableFunction;
+
+import org.apache.flink.api.connector.source.Boundedness;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
+import org.apache.flink.streaming.api.environment.CheckpointConfig;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.sink.DiscardingSink;
+import org.apache.flink.streaming.api.operators.StreamSource;
+
+import javax.annotation.Nullable;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Build for unaware-bucket table flink compaction job.
+ *
+ * <p>Note: This compaction job class is only used for unaware-bucket compaction, at start-up, it
+ * scans all the files from the latest snapshot, filter large file, and add small files into memory,
+ * generates compaction task for them. At continuous, it scans the delta files from the follow-up
+ * snapshot. We need to enable checkpoint for this compaction job, checkpoint will trigger committer
+ * stage to commit all the compacted files.
+ */
+public class UnawareBucketCompactionTopologyBuilder implements Serializable {
+
+    private static final String COMPACTION_COORDINATOR_NAME = "Compaction Coordinator";
+    private static final String COMPACTION_WORKER_NAME = "Compaction Worker";
+    private static final String COMPACTION_COMMITTER_NAME = "Compaction Committer";
+
+    private transient StreamExecutionEnvironment env;
+    private final String tableIdentifier;
+    private final AppendOnlyFileStoreTable table;
+    @Nullable private List<Map<String, String>> specifiedPartitions = null;
+    private boolean isContinuous = false;
+
+    public UnawareBucketCompactionTopologyBuilder(
+            StreamExecutionEnvironment env,
+            String tableIdentifier,
+            AppendOnlyFileStoreTable table) {
+        this.env = env;
+        this.tableIdentifier = tableIdentifier;
+        this.table = table;
+    }
+
+    public void withContinuousMode(boolean isContinuous) {
+        this.isContinuous = isContinuous;
+    }
+
+    public void withPartitions(List<Map<String, String>> partitions) {
+        this.specifiedPartitions = partitions;
+    }
+
+    public void build() {
+
+        // build source from UnawareSourceFunction
+        DataStreamSource<AppendOnlyCompactionTask> source = buildSource();
+
+        // from source, construct the full flink job
+        sinkFromSource(source);
+    }
+
+    private DataStreamSource<AppendOnlyCompactionTask> buildSource() {
+        UnawareBucketSourceFunction source = new UnawareBucketSourceFunction(table);
+        long scanInterval = table.coreOptions().continuousDiscoveryInterval().toMillis();
+        source.withStreaming(isContinuous);
+        source.withScanInterval(scanInterval);
+        source.withFilter(getPartitionFilter());
```

Comment:
UnawareBucketSourceFunction don't need to a builder pattern, parameters can be passed by constructor.